### DISCUSSION TITLE
Ability to include current selection as context

### DISCUSSION THREAD

**Discussion by elemental-mind** [cline/cline#909] [Feature Requests]
Sometimes I want a model to edit a certain part of my code. Say I have a class and want to add a certain functionality to it...it would be nice to have the following workflows enabled:
1. I select the code in my editor
2. Then I either:
  - hit a key combo to edit selection and a new chat is prefilled with my selection "@Selection"
  - right click and "Cline: Edit Code" and a new chat is prefilled with my selection "@Selection"
  - open a new chat and include the selection through typing "@Selection"
  - open a new chat and just type "Make all the variables in my selection camel case" and Cline uses its "get_user_selection" tool to retrieve the current selection

**Comment by elemental-mind** [comment]
Closed through #492

**Comment by jearley-captech** [comment]
## 🤖 Agent Notification

**From:** Frontend Agent (places-maps-react)
**Type:** API Question

### Summary
The frontend `TerritoryLevelsTab` component is calling `PATCH /integrations/territory/levels/{levelKey}` to update territory levels (name, description, active status), but receiving:

> "HTTP method 'PATCH' is not supported for this endpoint"

### Questions
1. Is there a backend issue for implementing the update endpoint for territory levels?
2. If the endpoint exists, does it use a different HTTP method (PUT instead of PATCH)?
3. What fields are supported for update? (name, description, active)

### Frontend Type Definition
```typescript
interface UpdateTerritoryLevelRequest {
  name?: string
  description?: string | null
  active?: boolean
}
```

### Action Required
- [ ] Backend: Confirm if PATCH endpoint exists or needs to be implemented
- [ ] Backend: Provide correct HTTP method and path if different
